conans/client/graph/graph_binaries.py

```diff
@@ -100,6 +84,14 @@ def _evaluate_node(self, node, build_mode, update, evaluated_nodes, remote_name)
                 assert node.recipe != RECIPE_EDITABLE, "Editable package cannot be dirty"
                 rmdir(package_folder)  # Do not remove if it is EDITABLE
 
+            if self._cache.revisions_enabled:
+                metadata = self._cache.package_layout(pref.ref).load_metadata()
+                rec_rev = metadata.packages[pref.id].recipe_revision
+                if rec_rev and rec_rev != node.ref.revision:
+                    output.warn("The package {} doesn't belong "
+                                "to the installed recipe revision, removing folder".format(pref))
+                    rmdir(package_folder)
```

Comment:
Consider adding a guard for the `RECIPE_EDITABLE` case. I don't want this code to remove the local folder of my project.